components/raftstore/src/store/peer.rs

```diff
@@ -704,6 +726,326 @@ pub enum UnsafeRecoveryState {
     Destroy(UnsafeRecoveryExecutePlanSyncer),
 }
 
+// #[RaftstoreCommon], copied from Peer
+/// Peer's trait
+pub trait RaftPeer<EK, ER>
+where
+    EK: KvEngine,
+    ER: RaftEngine,
+{
+    type Callback: ErrorCallback;
+
+    fn region(&self) -> &metapb::Region;
+    fn peer_id(&self) -> u64;
+    fn is_splitting(&self) -> bool;
+    fn is_merging(&self) -> bool;
+    fn is_leader(&self) -> bool;
+    fn term(&self) -> u64;
+    fn store_commit_index(&self) -> u64;
+    fn leader_id(&self) -> u64;
+    fn region_id(&self) -> u64;
+    fn tag(&self) -> &String;
+    fn txn_ext(&self) -> Arc<TxnExt>;
+    fn read_progress(&self) -> Arc<RegionReadProgress>;
+    fn peer(&self) -> &metapb::Peer;
+    fn bucket_meta(&self) -> Option<Arc<BucketMeta>>;
+    fn txn_extra_op(&self) -> Arc<AtomicCell<TxnExtraOp>>;
+    fn mut_pending_reads(&mut self) -> &mut ReadIndexQueue<Self::Callback>;
+    fn mut_bcast_wake_up_time(&mut self) -> &mut Option<TiInstant>;
```

Comment:
We are increasing region size in v2, so hibernate regions will probably be removed. There is no need to wake up anyone.